eth/protocols/snap/sync.go

```diff
@@ -745,6 +772,12 @@ func (s *Syncer) loadSyncStatus() {
 			for _, task := range s.tasks {
 				task := task // closure for task.genBatch in the stacktrie writer callback
 
+				// Restore the completed storages
+				task.stateCompleted = make(map[common.Hash]struct{})
+				for _, hash := range task.StorageCompleted {
+					task.stateCompleted[hash] = struct{}{}
+				}
```

Comment:
You should probably set `StorageCompleted` to nil here to clean out that memory area and make it consistent with newly created tasks.

Reply:
Good catch